src/core_inner/cluster/simple.rs

```diff
@@ -97,26 +97,29 @@ impl Cluster for Quorum {
         .into_iter()
         .collect::<FuturesUnordered<_>>();
 
-        let task = reqs
-            .collect::<Vec<_>>()
-            .map(move |acc| {
-                let sup = acc
-                    .iter()
-                    .filter_map(|r| r.as_ref().err().map(ToString::to_string))
-                    .collect();
-
-                let r = acc.into_iter().find(Result::is_ok);
-                if let Some(answer) = r {
-                    match answer {
-                        Ok(ClusterResult { result: i, .. }) => Ok(i),
-                        Err(ClusterResult { result: i, .. }) => Err(i),
-                    }
-                } else {
-                    debug!("GET[{}] no success result", key);
-                    Err(BackendError::Failed(sup))
-                }
-            })
-            .boxed();
+        let task = async move {
+            let results = reqs.collect::<Vec<_>>().await;
+            let err_results = results
+                .iter()
+                .filter_map(|r| r.as_ref().err())
+                .collect::<Vec<_>>();
+            let errors = err_results
```

Comment:
errors only logs, should we do something useful with them?

Reply:
I've thought, that it is the task for the next refactoring PR. There are a lot of that kind of code